core/src/blocktree.rs

```diff
@@ -540,13 +546,20 @@ impl Blocktree {
         let mut prev_inserted_coding = HashMap::new();
         let mut prev_inserted_blob_datas = HashMap::new();
 
+        let mut erasure_config_opt = None;
+
         for blob_item in blobs {
             let blob = blob_item.borrow();
             assert!(blob.is_coding());
 
+            match erasure_config_opt {
+                Some(config) => assert_eq!(blob.erasure_config(), config),
```

Comment:
this should induce an error, not an assert!()

Reply:
This was to make sure that peer is not switching erasure config on the fly. I can change this to an error. Should we update to the new config or keep the old config?

Reply:
slashing condition ;)  invalidate the slot and put a TODO; slash this guy

Reply:
👍 
Also, agree that we shouldn't assert!, otherwise external network events can trigger a network crash